### ISSUE TITLE
Reading expiration dates (iOS v5.0)

### ISSUE THREAD

**Issue by Natelegreat1** [card-io/card.io-iOS-SDK#92]
I heard that v5.0+ can supposedly detect expiration dates on iOS.
Is there something special to enable beyond what I had in v4.* ? Or should it just start working?
Currently, it is still only reading the card number... :(

**Comment by dgoldman-ebay** [comment]
@Natelegreat1 card.io 5.0 is our first release which attempts to scan expiration dates. Expiry-scanning is definitely a work in progress, but we feel that it's currently good enough to release to developers while we continue to work on it.

So far our impression is that there are many cards for which this version usually reads the expiry correctly, many cards for which it never succeeds, and many cards for which its performance varies considerably depending on lighting conditions etc. Part of our reason for releasing now is to collect feedback such as yours.

In fact, let's keep this github issue open as a place where others can report their impressions over the coming weeks.

**Comment by Natelegreat1** [comment]
Thanks! I kept trying with other cards. I guess my first try was just unlucky. 
So far...
Working : BMO SPC MC, BMO Enterprise MC, Costco AMEX
No luck : MBNA HABS MC, SCENE VISA, AMEX (frosted white), Scotiabank ScotiaGold, TD Visa

**Comment by mitchellporter** [comment]
I've tested this on both a Visa credit card and a Mastercard debit card. Both seem to work about 90% of the time

**Comment by dgoldman-ebay** [comment]
Thanks for the additional data points, @mitchellporter !

**Comment by debo2372** [comment]
Does the sample Swift app automatically scan the expiry date as well? It doesnt seem to be working on my VISA card.

**Comment by dgoldman-ebay** [comment]
@debo2372 yes, the Swift sample app behaves just like the ObjC sample app in this regard. As noted above, expiry-scanning does not currently work for all cards.

**Comment by Vijayakumr** [comment]
Hi @dgoldman-ebay. i am using card.io in android. its scanning card number exactly sometimes expiry date also i am getting & one question whether its scanning Card on name? is this possible?

**Comment by FeminaErnest** [comment]
We didn't get any update on this so far. So... scanning card on name is in progress I guess... 
http://stackoverflow.com/questions/16844028/reading-card-holders-name-by-card-io-library @dgoldman-ebay Am I correct Dave ?

**Comment by dgoldman-ebay** [comment]
@Vijayakumr this issue is about expiration dates. Questions regarding other features, such as scanning cardholder names, should be addressed in their own issues.

That said, please see https://github.com/card-io/card.io-iOS-SDK/issues/111.

**Comment by Vijayakumr** [comment]
Thanks @FeminaErnest,@dgoldman-ebay

**Comment by CaptainScavo** [comment]
My test results on Expiry Date scanning. Tested 5.2.2 on iPhone 6:

Expiry date scans - Cap1 Quicksilver
Expiry date doesn't scan- Wells Fargo VISA (date is MM/DD/YY format), Alliant VISA debit card, Bluebird card

**Comment by peacechen** [comment]
Expiration not scanning for these cards on iOS with CardIO 5.4.1.
- BoA MC (red background)
- BoA Visa (red background)
- DCU Visa (flat, which as expected doesn't scan)
- Citi MC (blue background)
- PlainsCapitalBank MC (red background)

Works for:
- Discover (silver background)
- BoA Visa (blue background)

It's interesting that the blue background BoA card recognizes the expiry, but the red background BoA cards don't.  I've tried multiple red BoA cards.  Also the PlainsCapitalBank MC with a red background fails.  The expiration dates are in the same relative positions between the red & blue cards.

Update: Tested outdoors to get as much light exposure as possible, and the red BoA card got the expiry ~50% of the time.

Back indoors, I found that rotating the tablet from portrait (our normal operating mode) to landscape increased the expiry recognition from 0% to ~33% (very rough figures).

Would there by any particular reason that landscape offers better expiry recognition?

**Comment by JoshuvaGeorge03** [comment]
is there any update on the issue? is this fixed? I was getting the card number but not getting the expiry date....
